File: pi_controller.cpp
#include <utility>
#include "pi_controller.h"
#include "Eigen/Eigen"

pi_controller::pi_controller(double p_arg, double i_arg){
    p = p_arg;
    i = i_arg;
}

double pi_controller::control(Matrix<2,1 double> e){
    double output = p * e(0,0) + i * e(1,0);
    if(output < 0.) return 0.;
    if(output > 0.125) return 0.125;
    return output;
}

File: pi_controller.m
classdef pi_controller
    properties
        p;
        i;
    end
    
    methods
        
        function obj = pi_controller(p,i)
            obj.p = p;
            obj.i = i;
        end
        
        function U_k = control(obj,e)
            output = obj.p*e(1)+obj.i*e(2);
            
            if(output<0)
                U_k = 0;
            elseif(output>0.125)
                U_k = 0.125; 
            else
                U_k = output;
            end
        end
        
    end
end